File: ListViewApp/app/src/main/java/com/example/listviewapp/MainActivity.java
package com.example.listviewapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    ArrayList<String> al;
    ListView list;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        list=findViewById(R.id.list);
        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Toast.makeText(parent.getContext(),"tutyio",Toast.LENGTH_LONG);
            }
        });
        al=new ArrayList<>();
        al.add("faster ai & analytics");
        al.add("faster Finacial services");
        al.add("faster Digital Services");
        al.add("faster Saas products");
        al.add("faster Holding");
        al.add("faster ai & analytics");
        al.add("faster Finacial services");
        al.add("faster Digital Services");
        al.add("faster Saas products");
        al.add("faster Holding");
        al.add("faster ai & analytics");
        al.add("faster Finacial services");
        al.add("faster Digital Services");
        al.add("faster Saas products");
        al.add("faster Holding");
        ArrayAdapter<String> ad=new ArrayAdapter(this, android.R.layout.simple_expandable_list_item_1,al);
        list.setAdapter(ad);
    }
}

File: SpinnerApp/app/src/main/java/com/example/spinnerapp/MainActivity.kt
package com.example.spinnerapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.Spinner
import android.widget.Toast
import androidx.core.view.get
import java.util.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var spinner = findViewById<Spinner>(R.id.spinner)
        val array = arrayOf("bob", "pnb", "boi", "sbi", "obc")
        val arrayAdapter: ArrayAdapter<String> =
            ArrayAdapter(applicationContext, android.R.layout.simple_list_item_1, array)
        arrayAdapter.setDropDownViewResource(android.R.layout.simple_list_item_1)
        spinner.adapter = arrayAdapter
        spinner.onItemSelectedListener = object : OnItemClickListener,
            AdapterView.OnItemSelectedListener {
            override fun onItemClick(
                parent: AdapterView<*>?,
                view: View?,
                position: Int,
                id: Long
            ) {
                Toast.makeText(applicationContext,spinner.get(position).toString(),Toast.LENGTH_SHORT).show()

            }

            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View?,
                position: Int,
                id: Long
            ) {
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
                TODO("Not yet implemented")
            }

        }
    }

}


